lib/graphql/schema/printer.rb

```diff
@@ -74,277 +85,37 @@ def self.print_schema(schema, **args)
 
       # Return a GraphQL schema string for the defined types in the schema
       def print_schema
-        directive_definitions = warden.directives.map { |directive| print_directive(directive) }
-
-        printable_types = warden.types.reject(&:default_scalar?)
-
-        type_definitions = printable_types
-          .sort_by(&:name)
-          .map { |type| print_type(type) }
-
-        [print_schema_definition].compact
-                                 .concat(directive_definitions)
-                                 .concat(type_definitions).join("\n\n")
+        print(@document)
       end
 
       def print_type(type)
-        TypeKindPrinters::STRATEGIES.fetch(type.kind).print(warden, type)
+        node = @document_from_schema.build_object_type_node(type)
+        print(node)
       end
 
-      private
-
-      # By default, these are included in a schema printout
-      IS_USER_DEFINED_MEMBER = ->(member) {
-        case member
-        when GraphQL::BaseType
-          !member.introspection?
-        when GraphQL::Directive
-          !member.default_directive?
+      def print_directive(directive)
+        if directive.name == "deprecated"
+          print_deprecated(directive)
         else
-          true
+          super
         end
-      }
+      end
 
-      private_constant :IS_USER_DEFINED_MEMBER
+      def print_deprecated(directive)
```

Comment:
Can this method be private? Or, better yet, can this code be inlined into `print_directive`? (I couldn't see any usages of it, but maybe I overlooked them!)

Reply:
good point, fixed!